lib/linguist/heuristics.rb

```diff
@@ -17,9 +17,8 @@ def self.call(blob, candidates)
       data = blob.data
 
       @heuristics.each do |heuristic|
-        if heuristic.matches?(blob.name)
-          languages = Array(heuristic.call(data))
-          return languages if languages.any? || languages.all? { |l| candidates.include?(l) }
```

Comment:
`languages.any? || languages.all? { … }` is always true; if `languages.any?` is false, then `languages.all? { … }` will always return true (because `.any?` on an empty list is always true). So the `if` was always happening.